src/main/java/org/mskcc/cmo/messaging/impl/NATSPublishingFileUtil.java

```diff
@@ -13,22 +13,39 @@
 @Component
 public class NATSPublishingFileUtil implements FileUtil {
     @Value("${metadb.publishing_failures_filepath}")
-    private String filePath;
+    private String publishingFailuresFilepath;
 
-    public boolean exists() {
-        File f = new File(filePath);
-        return f.exists();
-    }
+    private static final String PUB_FAILURES_FILE_HEADER = "DATE\tTOPIC\tMESSAGE\n";
 
     @Override
     public void savePublishFailureMessage(String topic, String message) throws IOException {
-        if (!exists()) {
-            File f = new File(filePath);
+        BufferedWriter fileWriter = getOrCreateFileWriterWithHeader(publishingFailuresFilepath,
```

Comment:
Why special case the call to write publish failure messages?

Reply:
That would be the default file writer whereas the other writer would be for implementing some other message detail logging like we're planning for the consistency checker

Reply:
the default publishing failure writer is also called by the NATSGatewayImpl so it's not something we use directly or explicitly in any of the clients